### PR TITLE
Fix migration failure

### PR DESCRIPTION
**Pull request by alukach**
Fix bug where migration fails if no spatial units are associated with project.

As mentioned in [#1716](https://github.com/Cadasta/cadasta-platform/pull/1716#issuecomment-324097777).

### Proposed changes in this pull request

Use `COALESCE` to fallback to `0` if `sum(<spatial_unit_areas>) == null`, in the event that a Project has no associated SpatialUnit instances.

### When should this PR be merged

Before releasing #1716.

### Risks

[List any risks that could arise from merging your pull request.]

-

### Follow-up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
  - [ ] Review 1
  - [ ] Review 2
- **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
  - [ ] Review 1
  - [ ] Review 2
- **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.
  - [ ] Review 1
  - [ ] Review 2

#### Functionality

- **Are all requirements met?** Compare implemented functionality with the requirements specification.
  - [ ] Review 1
  - [ ] Review 2
- **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 
  - [ ] Review 1
  - [ ] Review 2

#### Code

- **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
  - [ ] Review 1
  - [ ] Review 2
- **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
  - [ ] Review 1
  - [ ] Review 2
- **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 
  - [ ] Review 1
  - [ ] Review 2
- **Is the code documented sufficiently?** Large and complex classes, functions or methods must be annotated with comments following our [code-style guidelines](https://devwiki.corp.cadasta.org/Contributing%20Style%20Guide#documentation-and-comments).
  - [ ] Review 1
  - [ ] Review 2
- **Has the scalability of this change been evaluated?**
  - [ ] Review 1
  - [ ] Review 2
- **Is there a maintenance plan in place?**
  - [ ] Review 1
  - [ ] Review 2

#### Tests

- **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
  - [ ] Review 1
  - [ ] Review 2
- **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
  - [ ] Review 1
  - [ ] Review 2
- **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?
  - [ ] Review 1
  - [ ] Review 2

#### Security

- **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
  - [ ] Review 1
  - [ ] Review 2
- **Are all UI and API inputs run through forms or serializers?** 
  - [ ] Review 1
  - [ ] Review 2
- **Are all external inputs validated and sanitized appropriately?**
  - [ ] Review 1
  - [ ] Review 2
- **Does all branching logic have a default case?**
  - [ ] Review 1
  - [ ] Review 2
- **Does this solution handle outliers and edge cases gracefully?**
  - [ ] Review 1
  - [ ] Review 2
- **Are all external communications secured and restricted to SSL?**
  - [ ] Review 1
  - [ ] Review 2

#### Documentation

- **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
  - [ ] Review 1
  - [ ] Review 2
- **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
  - [ ] Review 1
  - [ ] Review 2
- **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki. 
  - [ ] Review 1
  - [ ] Review 2

### CODE REVIEW

**Comment by amplifi** [comment]
As discussed in #1716, this issue is being resolved at the location level with area values for locations without geometries defaulting to 0. That resolution is already submitted in #1728.

**Comment by seav** [comment]
No, we need to reopen this. I also encountered the null-constraint error @alukach ran into as mentioned in #1716 but I thought it was just me. Now that Anthony has explained the issue, the migration error makes perfect sense. This is a totally separate issue from the null-area locations bug. The issue is that the 0006 migration fails if the project has _no_ locations at all. So it does _not_ matter if location areas default to `null` or `0` because there are _no_ locations in the project.

**Comment by amplifi** [comment]
Ok, this will need to be changed to migration 0007 as there's already a 0006 in master.

**Comment by alukach** [comment]
@amplifi This is a bugfix to the 0006 migration, it is not a new migration. The issue is that organizations/0006 fails, so any subsequent migrations will not run (assuming the conditions described in https://github.com/Cadasta/cadasta-platform/pull/1716#issuecomment-324097777 are met).

**Comment by amplifi** [comment]
If this is a bugfix, it's required to include tests to prevent regression.

**Comment by alukach** [comment]
@amplifi Sure, I've updated the previously written test.

File: cadasta/organization/tests/test_migrations.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def test_migration(self):
         call_command('migrate', 'organization', self.migrate_from)
 
         org = Organization.objects.create(name='Test Org')
+        Project.objects.create(name='No Locations Proj', organization=org)
```
Comment (seav) [inline cadasta/organization/tests/test_migrations.py]: So the unit test failed because you need to specify unique `id`s for both projects.

Comment (alukach) [inline cadasta/organization/tests/test_migrations.py]: Ah, thanks. I just made the commit and closed my laptop. Will correct it tonight.

Comment (seav) [inline cadasta/organization/tests/test_migrations.py]: It might be better to use the existing `ProjectFactory` for cases like this.

**Comment by amplifi** [comment]
@seav Do the latest changes satisfy your review? If so, please update to approved. This is blocking for the staging release build.

**Pull request merged**